internal/worker/removal/manifold.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package removal
+
+import (
+	"context"
+
+	"github.com/juju/worker/v4"
+	"github.com/juju/worker/v4/dependency"
+
+	coredependency "github.com/juju/juju/core/dependency"
+	coreerrors "github.com/juju/juju/core/errors"
+	"github.com/juju/juju/core/logger"
+	"github.com/juju/juju/core/watcher"
+	removalservice "github.com/juju/juju/domain/removal/service"
+	"github.com/juju/juju/internal/errors"
+	"github.com/juju/juju/internal/services"
+)
+
+// DomainServices describes the service factory
+// required for working with entity removals.
+type DomainServices interface {
+	// Removal returns the service for managing entity removal.
+	Removal() *removalservice.WatchableService
+}
+
+// RemovalService describes the ability to watch
+// for and execute model entity removals.
+type RemovalService interface {
+	// WatchRemovals emits notifications for additions
```

Comment:
Please describe what the string of the notification represents. I'm guessing this supposed to be the ID of the removal job?